### PR TITLE
High Level Documentation

### PR DESCRIPTION
**Pull request by otto-AMA**
### Reason for this pull request

Adding a high-level (non-technical) overview of the data-cube architecture to the docs.  

### Proposed changes

- Add an `rst` formatted section under `Architecture` named High Level Architecture
- Add several diagrams
- Updated index file

### CODE REVIEW

**Review by GypsyBojangles — APPROVED** [review]

File: docs/ops/indexing.rst [hunk 1]
```diff
@@ -200,3 +200,124 @@ Once you have downloaded some data, it will need :ref:`metadata preparation
...
+scripts necessary for S3 indexing. The direct links are provided below
+since, at the time of this document, they are not all included in the
+latest release (1.6.1).
+
+::
+
+    cd ~/Datacube
+    mkdir -p S3_scripts
+    cd S3_scripts
+    wget https://raw.githubusercontent.com/opendatacube/datacube-core/develop/datacube/index/hl.py
```
Comment (Kirill888) [inline docs/ops/indexing.rst]: @otto-AMA 
This seems like a very fragile way of setting up prepare scripts? Why is this neccessary? You need datacube installed to perform indexing, why copying `hl.py` from github is needed?

Comment (Kirill888) [reply docs/ops/indexing.rst]: The script linked below imports hl from `datacube.index`, so it won't use local copy anyway, seems like leftover instruction from the past that is no longer relevant.

Comment (alexgleith) [reply docs/ops/indexing.rst]: Should we set up a new PR to resolve this now, while we're thinking about it?

Comment (Kirill888) [reply docs/ops/indexing.rst]: @alexgleith or @otto-AMA can either of you confirm that copying `hl.py` is not necessary (I believe it is not necessary) and if yes do new PR that removes unneeded instruction line from the docs? If this is being done in some docker/bash script file then that should be fixed too.

Comment (alexgleith) [reply docs/ops/indexing.rst]: Hey @otto-AMA do you think you could tackle this now, as it'll be easier while it's fresh!

**Pull request merged**